pandas/plotting/_matplotlib/core.py

```diff
@@ -288,6 +291,21 @@ def __init__(
 
         self.data = self._ensure_frame(self.data)
 
+        from pandas.plotting import plot_params
+
+        self.x_compat = plot_params["x_compat"]
+        if "x_compat" in self.kwds:
+            self.x_compat = bool(self.kwds.pop("x_compat"))
+
+    @final
+    def _is_ts_plot(self) -> bool:
+        # this is slightly deceptive
+        return not self.x_compat and self.use_index and self._use_dynamic_x()
+
+    @final
+    def _use_dynamic_x(self) -> bool:
+        return use_dynamic_x(self._get_ax(0), self.data)
+
```

Comment:
Those routines are not exclusive to LinePlot, as they seem related to any type of time series. `_is_ts_plot` is used in this PR to check if a boxplot is a time series.